### PR TITLE
Glutin version 0.30.0

### PR DESCRIPTION
**Pull request by kchibisov**
--
cc @i509VCB 
cc @chrisduerr 

If anyone has any concerns let me know.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,42 +1,15 @@
...
-- Fix crash in `glx::surface::Surface::set_swap_interval`.
-
-# Version 0.30.0-beta.2 (2022-09-03)
-
-- macOS: Fix memory leak in `Surface`.
-
-# Version 0.30.0-beta.1 (2022-09-03)
-
 - Replace `winit` dependency with `raw-window-handle`.
 - The Api is now built around `Display`, `Surface`, `Config`, and `Surface` for more info see crate documentation and examples.
```
Comment (MarijnS95) [inline CHANGELOG.md:7]:
```suggestion
- The Api is now built around `Display`, `Surface`, `Config`, and `Surface`. For more info see crate documentation and examples.
```

**Review by MarijnS95 — APPROVED** [review]
Lovely!

Would the `winit` removal warrant a `1.0.0` release, finally? Or do you want to let it boil for a little while longer? Does this **entire codebase refactor** need a bit more front-and-center attention in the changelog?

**Comment by kchibisov** [comment]
>Would the winit removal warrant a 1.0.0 release, finally? Or do you want to let it boil for a little while longer? Does this entire codebase and API refactor need a bit more front-and-center attention in the changelog?

I think this needs more battle testing, but yeah, `glutin` is not far from potential `v1.0.0` release. Though I'd wait for `raw-window-handle v1.0.0` which should be soon as well.

It would also be nice to see major consumers moving to `glutin 0.30`, like `egui`, `iced`, etc.

There're some api changes that we still need to make, but most of them are non-breaking. For example it would be nice to expose loaded libraries is some form and ability to create display from `raw` pointer. Like from raw `EGLDisplay`. I also need `GAT`s for some stuff internally, which should be soon, so I can remove redundant internal traits with generics...

It's also not clear what to do with `wasm` and `ios` support right now. It seems like both are possible to do, just need more attention, maybe with a leaner codebase there will be some attention to that problem and we have folks that care about it. Only time will tell. But having support for most platforms is already a good start.

>Does this entire codebase and API refactor need a bit more front-and-center attention in the changelog

I'm not sure how to put it in the CHANGLELOG, but I write something for `Release page` for sure. The thing is that we have entirely new crate in the end that has nothing to do with the old glutin other than name. So the CHANGELOG won't help here.

I do think that it would be nice to have more examples wrt complex handling of things, like multiwindow/multitheared stuff, context reset, but I don't have much time to write such examples myself (I ported alacritty with all of that stuff except multithreading, but **winit** on macOS is hard here, since I'm not sure how safe it is).

**Comment by MarijnS95** [comment]
I think you can just put it as the top-most, maybe even bold changelog point (`- This version of Glutin has been rewritten from the ground up blah blah and the new API revolves around [the four new types you mention later]`).

**Comment by kchibisov** [comment]
I guess it's not that a bad idea.

**Pull request merged**